src/main/kotlin/de/livepoll/api/util/websocket/SubscribeListener.kt

```diff
@@ -12,27 +13,45 @@ import org.springframework.web.socket.messaging.SessionSubscribeEvent
 
 @Component
 class SubscribeListener(
-    private val messagingTemplate: SimpMessageSendingOperations,
-    private val pollRepository: PollRepository,
-    private val pollItemService: PollItemService
+        private val messagingTemplate: SimpMessageSendingOperations,
+        private val pollRepository: PollRepository,
+        private val pollItemService: PollItemService,
+        private val webSocketService: WebSocketService
 ) : ApplicationListener<SessionSubscribeEvent> {
 
     @Transactional
     override fun onApplicationEvent(event: SessionSubscribeEvent) {
-        if( event.message.headers["simpDestination"].toString().contains("poll")){
-            val slug = event.message.headers["simpDestination"].toString().split("/").last()
+        val destination = event.message.headers["simpDestination"].toString()
+        if (destination.contains("poll")) {
+            val slug = destination.split("/").last()
             val poll = pollRepository.findBySlug(slug)
+            val url = "/v1/websocket/poll/$slug"
             if (poll == null) {
+                messagingTemplate.convertAndSendToUser(event.user!!.name, url, "{\"error\":\"Error\"}")
                 throw ResponseStatusException(HttpStatus.NOT_FOUND)
             } else {
-                val url = "/v1/websocket/poll/$slug"
                 if (poll.currentItem == null) {
                     messagingTemplate.convertAndSendToUser(event.user!!.name, url, "{\"pollId\":${poll.id}}")
+                    throw ResponseStatusException(HttpStatus.NOT_FOUND)
                 } else {
                     val pollItemDto = pollItemService.getPollItem(poll.currentItem!!)
                     messagingTemplate.convertAndSendToUser(event.user!!.name, url, pollItemDto)
                 }
             }
+        } else if (destination.contains("presentation")) {
+            val pollId = destination.split("/").last().toLong()
+            val url = "/v1/websocket/presentation/${pollId}"
+            pollRepository.findById(pollId).orElseThrow {
+                messagingTemplate.convertAndSendToUser(event.user!!.name, url, "{\"error\":\"Error\"}")
```

Comment:
Maybe we could extract that error into another method to avoid duplication?!

Reply:
Good idea. I did that